Notesnook.API/Controllers/InboxController.cs

```diff
@@ -115,5 +128,59 @@ public async Task<IActionResult> DeleteApiKeyAsync(string apiKey)
                 return BadRequest(new { error = ex.Message });
             }
         }
+
+        [HttpGet("public-encryption-key")]
+        [Authorize(Policy = InboxApiKeyAuthenticationDefaults.AuthenticationScheme)]
+        public async Task<IActionResult> GetPublicKeyAsync()
+        {
+            var userId = User.FindFirstValue("sub");
+            try
+            {
+                var userSetting = await UserSetting.FindOneAsync(u => u.UserId == userId);
+                if (string.IsNullOrWhiteSpace(userSetting?.InboxKeys?.Public))
+                {
+                    return BadRequest(new { error = "Inbox public key is not configured." });
+                }
+                return Ok(new { key = userSetting.InboxKeys.Public });
+            }
+            catch (Exception ex)
+            {
+                await Slogger<InboxController>.Error(nameof(GetPublicKeyAsync), "Couldn't get user's inbox's public key.", userId, ex.ToString());
+                return BadRequest(new { error = ex.Message });
+            }
+        }
+
+        [HttpPost("items")]
+        [Authorize(Policy = InboxApiKeyAuthenticationDefaults.AuthenticationScheme)]
+        public async Task<IActionResult> CreateInboxItemAsync([FromBody] SyncItem request)
+        {
+            var userId = User.FindFirstValue("sub");
+            try
+            {
+                if (request.Algorithm != "xsal-x25519-7")
+                {
+                    return BadRequest(new { error = "Only xsal-x25519-7 algorithm is supported." });
+                }
+                if (string.IsNullOrWhiteSpace(request.Cipher))
+                {
+                    return BadRequest(new { error = "Inbox item cipher is required." });
+                }
+                if (request.Length <= 0)
+                {
+                    return BadRequest(new { error = "Valid inbox item length is required." });
+                }
+
+                request.ItemId = Nanoid.Generate(size: 24);
```

Comment:
missed this previously, `ItemId` shouldn't be null

Reply:
This should be set from the `inbox-api-server` instead of setting it here.